text/0000-more-env-info.md

```diff
@@ -0,0 +1,58 @@
+- Feature Name: more-env-info
+- Start Date: 2017-08-09
+- RFC PR:
+- Rust Issue:
+
+# Summary
+[summary]: #summary
+
+Introduce more potentially useful environment-related functions to `std::env`.
+
+# Motivation
+[motivation]: #motivation
+
+Programs sometimes need to know about their working environment to do their job properly. For example, linebreak convention differs for Windows and *NIX. Such discrepancy can lead to problems easily, especially when a program needs to communicate with aged third-party libraries.
+
+As a system programming language, it would be good for Rust to know the system it works on.
+
+# Detailed design
+[design]: #detailed-design
+
+This RFC would like to introduce the following functions to `std::env`:
+
+```rust
+/// Platform architecture.
+/// e.g. `arm`, `x86_64` and `i686`.
+pub fn arch() -> String;
+/// Conventional linebreak of current platform.
+pub fn linebreak() -> String;
+/// Current operating system.
+/// e.g. `windows`, `linux` and `darwin`.
+pub fn os() -> String;
+/// Word size in bits the program has been compiled into.
+/// Commonly 32 or 64.
+pub fn word_size() -> u32;
```

Comment:
@fstirlitz If we use https://github.com/rust-lang/rfcs/issues/1339#issuecomment-262201151, the `usize` on x32 should be 32-bit. If it was 64-bit, the pointer-cast `(x: usize) as *const T` would not be a one-to-one mapping, which existing code do rely on.

A separate integer type would require a change of the provided `cfg` flags (e.g. `#[cfg(target_feature = "64-bit register")] type ifast = i64;`, which is out-of-scope for this RFC.